go/integTest/connection_test.go

```diff
@@ -14,6 +14,94 @@ import (
 	"github.com/valkey-io/valkey-glide/go/v2/internal/interfaces"
 )
 
+// createValkeyCluster creates a new ValkeyCluster with default settings
+func createValkeyCluster(clusterMode bool) (*ValkeyCluster, error) {
+	shardCount := 1
+	if clusterMode {
+		shardCount = 3
+	}
+	return NewValkeyCluster(false, WithClusterMode(clusterMode), WithShardCount(shardCount), WithReplicaCount(1))
+}
+
+// getClientListOutputCount parses CLIENT LIST output and returns the number of clients
+func getClientListOutputCount(output interface{}) int {
+	if output == nil {
+		return 0
+	}
+
+	var text string
+	switch v := output.(type) {
+	case []byte:
+		text = string(v)
+	case string:
+		text = v
+	default:
+		return 0
+	}
+
+	if text = strings.TrimSpace(text); text == "" {
+		return 0
+	}
+
+	return len(strings.Split(text, "\n"))
+}
+
+// getClientCount returns the number of connected clients
+func getClientCount(ctx context.Context, client interfaces.BaseClientCommands) (int, error) {
+	if clusterClient, ok := client.(interfaces.GlideClusterClientCommands); ok {
+		// For cluster client, execute CLIENT LIST on all nodes
+		result, err := clusterClient.CustomCommandWithRoute(ctx, []string{"CLIENT", "LIST"}, config.AllNodes)
+		if err != nil {
+			return 0, err
+		}
+
+		// Result will be a map with node addresses as keys and CLIENT LIST output as values
+		totalCount := 0
+		for _, nodeOutput := range result.MultiValue() {
+			totalCount += getClientListOutputCount(nodeOutput)
+		}
+		return totalCount, nil
+	}
+
+	// For standalone client, execute CLIENT LIST directly
+	glideClient := client.(interfaces.GlideClientCommands)
+	result, err := glideClient.CustomCommand(ctx, []string{"CLIENT", "LIST"})
+	if err != nil {
+		return 0, err
+	}
+	return getClientListOutputCount(result), nil
+}
+
+// getExpectedNewConnections returns the expected number of new connections when a lazy client is initialized
+func getExpectedNewConnections(ctx context.Context, client interfaces.BaseClientCommands) (int, error) {
+	if clusterClient, ok := client.(interfaces.GlideClusterClientCommands); ok {
+		// For cluster, get node count and multiply by 2 (2 connections per node)
+		result, err := clusterClient.CustomCommand(ctx, []string{"CLUSTER", "NODES"})
+		if err != nil {
+			return 0, err
+		}
+
+		var nodesInfo string
+		switch v := result.SingleValue().(type) {
+		case []byte:
+			nodesInfo = string(v)
+		case string:
+			nodesInfo = v
+		default:
+			nodesInfo = ""
+		}
+
+		if nodesInfo = strings.TrimSpace(nodesInfo); nodesInfo == "" {
+			return 0, nil
+		}
+
+		return len(strings.Split(nodesInfo, "\n")) * 2, nil
```

Comment:
probably
```suggestion
		return len(strings.Split(nodesInfo, "\r\n")) * 2, nil
```
RESP uses `\r\n` as a line separator

Reply:
I check with other clients and it is always \n. I tried to change but broke everything.
I debug the tests on python, java and go and the results are the same with just \n